city_scrapers/spiders/alle_port_authority.py

```diff
@@ -0,0 +1,190 @@
+        needs.
+        """
+        atable = self._build_datatable(response)
+        for row in atable:
+            data = {
+                'timezone': self.timezone,
+                '_type': 'event',
+                'name': self._parse_name(row),
+                'event_description': self._parse_description(row),
+                'classification': self._parse_classification(row),
+                'start': self._parse_start(row),
+                'end': self._parse_end(row),
+                'all_day': self._parse_all_day(row),
+                'location': self._parse_location(row),
+                'documents': self._parse_documents(row),
+                'sources': self._parse_sources(row),
+            }
+
+            data['status'] = self._generate_status(data)
+            data['id'] = self._generate_id(data)
+
+            yield data
+
+    def _parse_name(self, item):
+        """
+        Parse or generate event name.
+        """
+        return item[0]
+
+    def _parse_description(self, item):
+        """
+        Parse or generate event description.
+        """
+        return ''
+
+    def _parse_classification(self, item):
+        """
+        Differentiate board meetings from public hearings.
+        """
+        meeting_title = item[0].lower()
+        if 'committee' in meeting_title:
+            return COMMITTEE
+        return BOARD
+
+    def _parse_start_datetime(self, item):
+        """
+        Parse the start datetime.
+        """
+        if 'cancel' in item[2].lower():
+            return ''
+
+        if not item[1].strip():
+            if 'stakeholder' in item[0].lower():
+                time = '8:30 a.m.'
+            if 'performance oversight' in item[0].lower():
+                time = '9:00 a.m.'
+            else:
+                time = '9:30 a.m.'
+        else:
+            time = item[1]
+        date = ('{year} {date}'
+                .format(year=self.event_year, date=item[2]))
+
+        time_string = '{0} {1}'.format(date, time)
+        return (parse(time_string))
+
+    def _parse_start(self, item):
+        datetime_obj = self._parse_start_datetime(item)
+        if not datetime_obj:
+            return ''
+        return {
+            'date': datetime_obj.date(),
+            'time': datetime_obj.time(),
+            'note': ''
+        }
+
+    def _parse_end(self, item):
+        """
+        No end date listed. Estimate 3 hours after start time.
+        """
+        datetime_obj = self._parse_start_datetime(item)
+        if not datetime_obj:
+            return ''
+        return {
+            'date':  datetime_obj.date(),
+            'time': ((datetime_obj + timedelta(hours=3))
+                     .time()),
+            'note': 'Estimated 3 hours after start time'
+        }
+
+    def _parse_all_day(self, item):
+        """
+        Parse or generate all-day status. Defaults to False.
+        """
+        return False
+
+    def _parse_location(self, item):
```

Comment:
This works, but it might be good to check to make sure a part of this text is in the page (maybe just the ZIP and room?) and throw an error if we can't find it. That way we'll know if the meeting location ever changes